src/FriendLibraryPage.tsx

```diff
@@ -32,83 +34,108 @@ const FriendLibraryPage: React.FC = () => {
         console.log(error);
       })
       .finally(() => {
-        setBooksAreLoaded(true);
+        setIsLoaded(true);
       });
+  }, [id]);
+
+  const [modalOpen, setModalOpen] = useState(false);
+  const [isNewRequest, setIsNewRequest] = useState(true);
+  const [selectedLoan, setSelectedLoan] = useState<Loan | LoanRequest>({
+    book_id: "",
+    requester_contact: "",
+    status: "pending",
+  });
+
+  const handleRequest = useCallback((bookID: string) => {
+    setIsNewRequest(true);
+    setSelectedLoan({
+      book_id: bookID,
+      requester_contact: "",
+      status: "pending",
+    });
+    setModalOpen(true);
   }, []);
 
-  useEffect(() => {
-    fetch("http://paralibrary.digital/api/loans/requester", {
-      credentials: "include",
-    })
-      .then((res) => {
-        return res.json();
-      })
-      .then(
-        (result) => {
-          setRequests(result);
+  const updateDatabase = useCallback(
+    (loan: LoanRequest) => {
+      fetch("http://paralibrary.digital/api/loans", {
+        method: "POST",
+        credentials: "include",
+        headers: {
+          "Content-Type": "application/json",
         },
-        (error) => {
-          console.log(error);
-        }
-      )
-      .catch((error) => {
-        console.log(error);
+        body: JSON.stringify(loan),
       })
-      .finally(() => {
-        setLoansAreLoaded(true);
-      });
-  }, []);
+        .then((res) => res.json())
+        .then((res: Loan) => {
+          setBooks(
+            books.map((book) =>
+              book.id !== loan.book_id ? book : { ...book, loan: res }
```

Comment:
I think this pattern's a good way to update local data only if the server sends back a positive response.